packages/storage/src/file.js

```diff
@@ -729,6 +729,13 @@ File.prototype.createResumableUpload = function(options, callback) {
  * Resumable uploads are automatically enabled and must be shut off explicitly
  * by setting `options.resumable` to `false`.
  *
+ * <p class="notice">
+ *   Resumable uploads have been known to cause performance issues. For each
+ *   file being uploaded we store the first 16kb of data, this can cause
+ *   considerable performance degredation when uploading a large number of
+ *   files.
+ * </p>
```

Comment:
Small and medium are loose terms which is why I threw out 10 MB. I don't think resumable uploads are even possible upstream if the file size is less than that, so wasting time caching information about the upload is where the performance hit comes from.